elide-integration-tests/src/test/java/com/yahoo/elide/tests/AnyPolymorphismIT.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2017, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.tests;
+
+import com.jayway.restassured.RestAssured;
+import com.yahoo.elide.core.HttpStatus;
+import com.yahoo.elide.utils.JsonParser;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import static org.hamcrest.Matchers.equalTo;
+
+public class AnyPolymorphismIT {
+    private static final String JSONAPI_CONTENT_TYPE = "application/vnd.api+json";
+    private final JsonParser jsonParser = new JsonParser();
+
+    @BeforeClass
+    public void setUp() {
+        //Create a tractor
+        RestAssured
+                .given()
+                .contentType(JSONAPI_CONTENT_TYPE)
+                .accept(JSONAPI_CONTENT_TYPE)
+                .body("{\"data\": {\"type\": \"tractor\", \"attributes\": {\"horsepower\": 102 }}}")
+                .post("/tractor")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED);
+
+        //Create a smartphone
+        RestAssured
+                .given()
+                .contentType(JSONAPI_CONTENT_TYPE)
+                .accept(JSONAPI_CONTENT_TYPE)
+                .body("{\"data\": {\"type\": \"smartphone\", \"attributes\": {\"type\": \"android\" }}}")
+                .post("/smartphone")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED);
+    }
+
+    @Test
+    public void testAny() {
```

Comment:
There are 4 different ways Elide generates HQL:
```
/someRootEntity - a query against a root entity
/someRootEntity?page[totals] - a query including page totals
/someRootEntity/24/anotherEntity - a query against a subcollection
/someRootEntity/24/anotherEntity?page[totals] - a query including page totals against a subcollection.
```
To really make sure this works, we should have a test for all of the above.

Reply:
Tests have been added. They did show one issue which I fixed.